File: example/plugins/com.adjust.sdk.purchase/ext/iOS/sdk/AdjustPurchase/ADJPConfig.m
//
//  ADJPConfig.m
//  AdjustPurchase
//
//  Created by Uglješa Erceg on 07/12/15.
//  Copyright © 2015 adjust GmbH. All rights reserved.
//

#import "ADJPConfig.h"
#import "ADJPLogger.h"

static const NSUInteger kAppTokenLength = 12;

@implementation ADJPConfig

#pragma mark - Object lifecycle

- (id)initWithAppToken:(NSString *)appToken
        andEnvironment:(NSString *)environment {
    self = [super init];

    if (!self) {
        return self;
    }

    _appToken = appToken;
    _environment = environment;
    self.logLevel = ADJPLogLevelInfo;

    return self;
}

#pragma mark - Public methods

- (BOOL)isValid:(NSString *)errorMessage {
    NSString *message;

    if (self.appToken == nil) {
        message = @"App token can't be nil";
        [ADJPLogger error:message];

        if (errorMessage != nil) {
            errorMessage = message;
        }

        return NO;
    }

    if ([self.appToken length] != kAppTokenLength) {
        message = @"App token must be 12 characters long";
        [ADJPLogger error:message];

        if (errorMessage != nil) {
            errorMessage = message;
        }

        return NO;
    }

    if (self.environment == nil) {
        message = @"Environment can't be nil";
        [ADJPLogger error:message];

        if (errorMessage != nil) {
            errorMessage = message;
        }

        return NO;
    }

    if ([self.environment isEqualToString:ADJPEnvironmentSandbox]) {
        [ADJPLogger assert:@"SANDBOX: AdjustPurchase is running in sandbox mode. Use this setting for testing. Don't forget to set the environment to `production` before publishing!"];
        return YES;
    } else if ([self.environment isEqualToString:ADJPEnvironmentProduction]) {
        [ADJPLogger assert:@"PRODUCTION: AdjustPurchase is running in production mode. Use this setting only for the build that you want to publish. Set the environment to `sandbox` if you want to test your app!"];
        return YES;
    } else {
        message = [NSString stringWithFormat:@"Unknown environment '%@'", self.environment];
        [ADJPLogger error:message];

        if (errorMessage != nil) {
            errorMessage = message;
        }

        return NO;
    }
}

@end

File: example/plugins/com.adjust.sdk.purchase/ext/Android/sdk/AdjustPurchase/adjust_purchase/src/main/java/com/adjust/sdk/purchase/ADJPConfig.java
package com.adjust.sdk.purchase;

import java.util.Locale;

/**
 * Created by uerceg on 09/12/15.
 */
public class ADJPConfig {
    private String appToken;
    private String sdkPrefix;
    private String environment;
    private ADJPLogLevel logLevel;

    public ADJPConfig(String appToken, String environment) {
        this.appToken = appToken;
        this.environment = environment;
        this.logLevel = ADJPLogLevel.INFO;
    }

    public String getAppToken() {
        return this.appToken;
    }

    public String getEnvironment() {
        return this.environment;
    }

    public String getClientSdk() {
        if (this.sdkPrefix == null) {
            return ADJPConstants.SDK_VERSION;
        } else {
            return String.format(Locale.US, "%s@%s", sdkPrefix, ADJPConstants.SDK_VERSION);
        }
    }

    public ADJPLogLevel getLogLevel() {
        return this.logLevel;
    }

    public void setSdkPrefix(String sdkPrefix) {
        this.sdkPrefix = sdkPrefix;
    }

    public void setLogLevel(ADJPLogLevel logLevel) {
        this.logLevel = logLevel;
    }

    public boolean isValid() {
        if (this.appToken == null) {
            ADJPLogger.getInstance().error("Invalid app token");
            return false;
        }

        if (this.appToken.length() != ADJPConstants.APP_TOKEN_SIZE) {
            ADJPLogger.getInstance().error("Invalid app token");
            return false;
        }

        if (this.environment == null) {
            ADJPLogger.getInstance().error("Invalid environment");
            return false;
        }

        if (this.environment.equalsIgnoreCase(ADJPConstants.ENVIRONMENT_SANDBOX) == false &&
                this.environment.equalsIgnoreCase(ADJPConstants.ENVIRONMENT_PRODUCTION) == false) {
            ADJPLogger.getInstance().error("Invalid environment");
            return false;
        }

        if (this.environment.equalsIgnoreCase(ADJPConstants.ENVIRONMENT_SANDBOX)) {
            ADJPLogger.getInstance().Assert("SANDBOX: AdjustPurchase is running in sandbox mode. " +
                    "Use this setting for testing. " +
                    "Don't forget to set the environment to `production` before publishing!");
            return true;
        }

        if (this.environment.equalsIgnoreCase(ADJPConstants.ENVIRONMENT_PRODUCTION)) {
            ADJPLogger.getInstance().Assert("PRODUCTION: AdjustPurchase is running in production mode. " +
                    "Use this setting only for the build that you want to publish. " +
                    "Set the environment to `sandbox` if you want to test your app!");
            return true;
        }

        return true;
    }
}
